packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -170,6 +177,15 @@ function reducer<T, C>(data: AsyncListState<T, C>, action: Action<T, C>): AsyncL
             items: action.type === 'loading' ? [] : data.items,
             abortController: action.abortController
           };
+        case 'update':
+          // TODO: Ask if this is appropriate, I want to cancel prior loads if the user is rapidly typing
+          // but "update" also covers a bunch of other actions (like item insertion)
+          // Was loading but update happened, so abort previous load.
+          data.abortController.abort();
+          return {
+            ...data,
+            ...action.updater(data)
```

Comment:
perhaps alternative is to overwrite the `setFilterText` that comes from `useListData` and classify it as a different action than `update` so I can have this logic only happen for when filterText is being changed

Reply:
I think you may need to overwrite that function and instead trigger a `loading` action, or perhaps a new action type. You may want to use a new action type to avoid the items being removed while loading (similar to `loadingMore` and `sorting`.